### PR TITLE
Added resource_info.go 

### PR DESCRIPTION
**Pull request by swiftdiaries**
Addressing issue: https://github.com/kubernetes-incubator/kube-arbitrator/issues/29.

### CODE REVIEW

**Comment by k82cn** [comment]
@swiftdiaries  , please sign cla first :).

**Comment by swiftdiaries** [comment]
@k82cn I have signed it, but for some reason the bot is not recognizing it.

**Comment by k82cn** [comment]
@swiftdiaries , please check detail to see what happened :).

File: pkg/schedulercache/resource_info.go [hunk 1]
```diff
@@ -16,5 +16,70 @@ limitations under the License.
...
+		Memory:   r.Memory,
+	}
+}
+
+var minMilliCPU float64 = 10
+var minMemory float64 = 10 * 1024 * 1024
+
+func NewResource(rl v1.ResourceList) *Resource {
+	cpu := rl[v1.ResourceCPU]
+	mem := rl[v1.ResourceMemory]
```
Comment (k82cn) [inline pkg/schedulercache/resource_info.go:54]: We need to check return value of map, some quest may only have cpu request.

File: pkg/schedulercache/resource_info.go [hunk 2]
```diff
@@ -16,5 +16,70 @@ limitations under the License.
...
+
+func (r *Resource) Add(rr *Resource) *Resource {
+	r.MilliCPU += rr.MilliCPU
+	r.Memory += rr.Memory
+	return r
+}
+
+func (r *Resource) Sub(rr *Resource) *Resource {
+	r.MilliCPU -= rr.MilliCPU
+	r.Memory -= rr.Memory
```
Comment (k82cn) [inline pkg/schedulercache/resource_info.go:74]: we need to check the value of each resource, negative memory does not make sense.

File: pkg/schedulercache/resource_info.go [hunk 3]
```diff
@@ -16,5 +16,70 @@ limitations under the License.
...
+	return r
+}
+
+func (r *Resource) Less(rr *Resource) bool {
+	return r.MilliCPU < rr.MilliCPU && r.Memory < rr.Memory
```
Comment (k82cn) [inline pkg/schedulercache/resource_info.go:79]: check nil

File: pkg/schedulercache/resource_info.go [hunk 4]
```diff
@@ -16,5 +16,70 @@ limitations under the License.
...
+}
+
+func (r *Resource) LessEqual(rr *Resource) bool {
+	return (r.MilliCPU < rr.MilliCPU || math.Abs(rr.MilliCPU-r.MilliCPU) < 0.01) &&
```
Comment (k82cn) [inline pkg/schedulercache/resource_info.go:83]: check nil

**Review by k82cn — CHANGES_REQUESTED** [review]
Thanks very much ! that's the basic function what we want. Just some suggestion on safe checking, please help to add comments & unit test

**Comment by k82cn** [comment]
BTW, please also squash/merge those commit into one.

**Comment by swiftdiaries** [comment]
Thank you for the review @k82cn :) I'll add changes and update accordingly. I'll also try to add comments and unit tests.

File: pkg/schedulercache/resource_info.go [hunk 5]
```diff
@@ -16,5 +16,70 @@ limitations under the License.
...
+}
+
+func EmptyResource() *Resource {
+	return &Resource{
+		MilliCPU: 0,
+		Memory:   0,
+	}
+}
+
+func CopyResource(r *Resource) *Resource {
```
Comment (jinzhejz) [inline pkg/schedulercache/resource_info.go:42]: suggest use `Clone()` as other xxx_info.go

**Comment by swiftdiaries** [comment]
Closing this PR because of CLA issues and a messed up git history. I'll reopen a new one and submit again.

**Pull request closed without merging**